streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovement.java

```diff
@@ -102,8 +111,16 @@ public int hashCode() {
                         sourceClientTasksIterator.remove();
                         statefulActiveTaskAssignment.get(destination).add(task);
                     } else {
+                        if (clientStates.get(destination).prevStandbyTasks().contains(task)
+                                && tasksToRemainingStandbys.get(task) > 0
```

Comment:
The only way we can ever hit this (task had this standby but remainingStandbys is 0) is if the configured `num.standby.replicas` is dynamically changed. But I believe we do allow that...right?

Reply:
Yes, we do.